src/pg/guc.cpp

```diff
@@ -0,0 +1,15 @@
+#include "pgduckdb/pg/guc.hpp"
+#include "pgduckdb/pgduckdb_utils.hpp"
+
+extern "C" {
+#include "postgres.h"
+#include "utils/guc.h"
+}
+
+namespace pgduckdb::pg {
+
+const char *
+GetConfigOption(const char *name, bool missing_ok, bool restrict_privileged) {
+	return PostgresFunctionGuard(::GetConfigOption, name, missing_ok, restrict_privileged);
+}
+} // namespace pgduckdb::pg
```

Comment:
I'm a bit on the fence since they're only used in one file, but should we move the `DefineCustomVariable` helpers from `pgduckdb.cpp` here?

Reply:
I think we should make all our functions in `src/pg/*.cpp` files be functions that are safe to call from C++. Since `DefineCustomVariable` are not, and don't really make sense to be, since they are called from `_PG_init`, I don't think that makes sense to move them. I'd like to move all our GUC definitions to a different file in general though.